### DISCUSSION TITLE
How does copilot X security differ from github copilot for business?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#68500] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

We are considering implementing Copilot chat and voice into our company.
Could you allow me to ask a question as there is no documentation available on security for chat and voice.

I have confirmed that github copilot for business is secure in terms of the following. github chat,github voice is the same?

- Are communications encrypted when prompted?
- Are code snippets stored on github?
- Does the github AI learn when the prompt is entered?

Thank you for your attention.

**Comment by Akash1134** [comment, marked as answer]
Hi @kushida-kaonavi ,

Please refer the [limitations of Copilot Chat](https://docs.github.com/en/copilot/github-copilot-chat/about-github-copilot-chat#limitations-of-github-copilot-chat) for details. 

For rest of your questions:

> Are communications encrypted when prompted?

**Encryption**: The transmitted data is encrypted in transit and at rest. 

**Access controls**: Access is strictly controlled. The data can only be accessed by (1) named GitHub personnel working on the GitHub Copilot team or on the GitHub platform health team, and (2) Microsoft personnel working on or with the GitHub Copilot team


GitHub Copilot Chat beta uses three types of data to provide the service:

- **User Engagement Data**: User Engagement Data is data about user interactions with the service and overall usage data, which includes pseudonymous identifiers. A pseudonymous identifier is like a code name for a person. It's a privacy tool allowing data collection without directly linking it to a specific individual. 
- **Contextual Prompts**:  A Contextual Prompt is the combination of a summary of contextual information gleaned from the user as well as the question submitted in the chat interface
- **Suggestions**: A Suggestion is one or more lines of proposed text returned to the GitHub Copilot Chat beta extension after a Contextual Prompt is received and processed by the AI-model. It is the output provided by the chat system to answer a user's query or engage in a conversation with the user. 

> Are code snippets stored on github?


The data handling details depend on whether you are using GitHub Copilot Chat beta as a GitHub Copilot for Business or GitHub Copilot for Individuals user and your settings if you are a Copilot for Individuals user. 

Contextual Prompts and Suggestions are not stored at rest for Copilot for Business users

**If you use Copilot for Individuals, your Contextual Prompts and Suggestions from your interactions with Copilot Chat beta will be retained unless you’ve disabled such collection** in your [current Copilot settings](https://github.com/settings/copilot) 

User Engagement Data is collected and retained from all users, regardless of product type, for 2 years.

> Does the github AI learn when the prompt is entered?

GitHub does not store Contextual Prompts or Suggestions from use of GitHub Copilot Chat beta for Copilot for Business users. GitHub only stores **Contextual Prompts and Suggestions** from use of Copilot Chat beta for GitHub Copilot for Individual users that have opted to enable such collection in their Copilot settings. Regardless, GitHub does not share that data with other users. 

Hope that helps!

**Reply by kikaragyozov** [reply]
@Akash1134 If I understood correctly, the technical preview for Github Copilot Chat for business **has telemetry involved**, specifically user engagement data, which cannot be disabled.

Such data relates to the user interactions with the service and overall usage data. Could you dive a little deeper on the "overall usage data" being covered? Theoretically, is it possible to divulge or **assume** what the user has written in the chat, and/or what the output of the service was, or assumed to be, based on the telemetry data being logged?

![Screenshot_1](https://github.com/community/community/assets/6695731/fd227aec-e8ea-42ef-9e6c-0b050a88047b)
